src/vs/workbench/browser/quickaccess.ts

```diff
@@ -73,27 +83,54 @@ export class EditorViewState {
 				state: getIEditor(activeEditorPane.getControl())?.saveViewState() ?? undefined,
 			};
 		}
+
+	}
+
+	/**
+	 * Open a transient editor such that it may be closed when the state is restored.
+	 * Note that, when the state is restored, if the editor is no longer transient, it will not be closed.
+	 */
+	async openTransientEditor(editor: IResourceEditorInput | ITextResourceEditorInput | IUntitledTextResourceEditorInput | IUntypedEditorInput, group?: IEditorGroup | GroupIdentifier | SIDE_GROUP_TYPE | ACTIVE_GROUP_TYPE | AUX_WINDOW_GROUP_TYPE): Promise<IEditorPane | undefined> {
+		editor.options = { ...editor.options, transient: true };
+		const openEditor = await this.editorService.openEditor(editor, group);
+		if (openEditor?.input) {
+			this.openedEditors.add(openEditor.input);
+		}
+		return openEditor;
 	}
 
-	async restore(shouldCloseCurrEditor = false): Promise<void> {
+	async restore(): Promise<void> {
 		if (this._editorViewState) {
 			const options: IEditorOptions = {
 				viewState: this._editorViewState.state,
 				preserveFocus: true /* import to not close the picker as a result */
 			};
-			if (shouldCloseCurrEditor) {
-				const activeEditorPane = this.editorService.activeEditorPane;
-				const currEditor = activeEditorPane?.input;
-				if (currEditor && currEditor !== this._editorViewState.editor && activeEditorPane?.group.isPinned(currEditor) !== true) {
-					await activeEditorPane.group.closeEditor(currEditor);
-				}
-			}
+			// close any transient editors that are still open and were opened by this instance of EditorViewState
+			const groups = this.editorService.visibleEditorPanes.map(group => group.group);
+
+			const closeEditorPromises: Promise<boolean> = Promise.resolve(true);
+			this.openedEditors.forEach(openedEditor => {
+				groups.forEach(group => {
+					if (group.contains(openedEditor) && group.isTransient(openedEditor)) {
+						closeEditorPromises.then(() => group.closeEditor(openedEditor));
+					}
+				});
+			});
+			await closeEditorPromises;
```

Comment:
Can we not use `IEditorService.closeEditors()` method here after we have checked that the editor is still transient? In addition, I would also check if the editor is dirty, otherwise this would trigger a confirmation dialog.

Reply:
Yeah its possible if an extension manages to modify the document or any other action that does not involve moving focus into the editor.

Reply:
I can add a check for now that the editor is not dirty.